internal/component/prometheus/operator/common/component_test.go

```diff
@@ -94,25 +96,25 @@ func TestRunExit(t *testing.T) {
 	ctx, cancelFunc := context.WithCancel(context.Background())
 	cmpRunExited := atomic.Bool{}
 	cmpRunExited.Store(false)
+
+	wg := &sync.WaitGroup{}
+	wg.Add(1)
 	go func() {
+		wg.Done()
```

Comment:
Not sure why `wg.Done()` is in the beginning of the goroutine? Does it guarantee anything?

Reply:
It ensures that we have entered the go routine, its still not perfect timing but hasnt failed in my testing.